### PR TITLE
Feature/df 1070 send anonymous contact in candidate

### PR DESCRIPTION
**Pull request by fabiangotzen**

### CODE REVIEW

File: candidate-app/src/main/resources/config/liquibase/changelog/20190611180000_spring_integration.xml [hunk 1]
```diff
@@ -0,0 +1,149 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
```
Comment (fabiangotzen) [inline candidate-app/src/main/resources/config/liquibase/changelog/20190611180000_spring_integration.xml]: alv-mail-sending-starter has a transitive dependency on alv-spring-integration-starter which needs these tables. 

Second part of this refactoring would be to fully integrate it, but will be handled in a different PR.

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/candidate/impl/mail/AnonymousContactMailPreparator.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        context.setVariable("subject", anonymousContactMessage.getSubject());
+        context.setVariable("personalMessage", StringUtils.defaultIfBlank(anonymousContactMessage.getPersonalMessage(), null));
+        context.setVariable("companyName", anonymousContactMessage.getCompanyName());
+        context.setVariable("phone", anonymousContactMessage.getPhone());
+        context.setVariable("email", anonymousContactMessage.getEmail());
+        prepareCompanyData(context, anonymousContactMessage.getCompany());
+        return context;
+    }
+
+    private static void prepareCompanyData(Context context, AnonymousContactMessageDTO.CompanyDTO company) {
```
Comment (birom) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/candidate/impl/mail/AnonymousContactMailPreparator.java]: I'd prefer not to use context as an argument. Return a map and set it to the conext:
Eg.:

```
 private static Map<String, Object> extractCompanyData(AnonymousContactMessageDTO.CompanyDTO company) {
        if (company == null) {
            return Collections.singletonMap("hasCompany", false);
        }

        Map<String, Object> companyData = new HashMap<>();
        companyData.put("hasCompany", true);
        companyData.put("name", company.getName());
        companyData.put("contactPerson", company.getContactPerson());
        companyData.put("address", prepareCompanyAddress(company));
        companyData.put("city", prepareCompanyCity(company));
        companyData.put("country", company.getCountry());

        return companyData;
    }
```

and then
```
 static Context prepareAnonymousContactMailContext(AnonymousContactMessageDTO anonymousContactMessage) {
        Locale locale = LocaleContextHolder.getLocale();
        Context context = new Context(locale);
        context.setVariable("subject", anonymousContactMessage.getSubject());
        context.setVariable("personalMessage", StringUtils.defaultIfBlank(anonymousContactMessage.getPersonalMessage(), null));
        context.setVariable("companyName", anonymousContactMessage.getCompanyName());
        context.setVariable("phone", anonymousContactMessage.getPhone());
        context.setVariable("email", anonymousContactMessage.getEmail());
        context.setVariables(extractCompanyData(anonymousContactMessage.getCompany()));

        return context;
    }
```

Comment (fabiangotzen) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/candidate/impl/mail/AnonymousContactMailPreparator.java]: fixed

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/candidate/impl/mail/AnonymousContactMailPreparator.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private static String extractCompanyStreetWithHouseNumber(AnonymousContactMessageDTO.CompanyDTO company) {
+        return joinStrings(company.getStreet(), company.getHouseNumber());
+    }
+
+    private static String extractCompanyCityWithZipCode(AnonymousContactMessageDTO.CompanyDTO company) {
+        return joinStrings(company.getZipCode(), company.getCity());
+    }
+
+    private static String joinStrings(String... values) {
+        return StringUtils.join(
```
Comment (birom) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/candidate/impl/mail/AnonymousContactMailPreparator.java]: Just some cosmetic: there is a collectror that joins a string
```
String s = Stream.of(values)
            .filter(StringUtils::isNotEmpty)
            .collect(Collectors.joining(COMMA_SEPERATOR))
```
